### ISSUE TITLE
Graphicsmagick Integration

### ISSUE THREAD

**Issue by meteorjs3000** [arunoda/meteor-up-legacy#308]
Dear all, may I suggest for the integration of Graphicsmagick and/or Imagemagick into mup? 

I believe many Meteor apps deal with images and GM/IM are two of the most widely used softwares for image manipulation. It'll be great to have GM/IM installation option apart from phantom.

**Comment by nate331** [comment]
+1

**Comment by miri-am** [comment]
+1

**Comment by kevohagan** [comment]
+1

**Comment by jamesgibson14** [comment]
I  like  it.

**Comment by rhyslbw** [comment]
:+1:

**Comment by arunoda** [comment]
Okay. We are on the track.
See: https://github.com/arunoda/meteor-up/tree/mupx#

With mupx you can choose which docker image to run as the meteor runtime. 
Currently it uses MeteorD, but you can use extend it add anything you want.

This is a planned feature and will be implement soon.

**Comment by rhyslbw** [comment]
@arunoda Is there an API for specifying the docker image?

**Comment by arunoda** [comment]
We'll have it soon.

**Comment by oliversd** [comment]
+1

**Comment by dangilkerson** [comment]
+1

**Comment by achtan** [comment]
ok, so how to currently install Graphicsmagick? i am using mupx

**Comment by nate331** [comment]
when you have a debian/ubuntu server: you log into the server via ssh and then install it via:
`sudo apt-get install imagemagick`
If you have setup your meteor/meteorup environment before already, you have to restart it with
`mup restart`

**Comment by achtan** [comment]
yes i did that, but this install gm/im to 'server' not to docker machine

**Comment by miri-am** [comment]
same problem like @achtan. See also #635

**Comment by mario589736** [comment]
This worked for me:

http://stackoverflow.com/questions/31901697/meteor-up-docker-and-graphicsmagick/31903981#31903981

**Comment by miri-am** [comment]
@mario589736. Yes, the approach you linked to worked for me. You can also add it manually using `docker exec -it containerName apt-get install graphicsmagick

**Comment by bompi88** [comment]
+1... 

 `docker exec -it containerName apt-get install graphicsmagick -y` did not work for me.

**Comment by samylaumonier** [comment]
You can install the package manualy (for now) :
`sudo apt-get install imagemagick`

**Comment by projectn00b** [comment]
Any word on this? I am trying to add ffmpeg.

**Comment by ignacy130** [comment]
+1; Nothing, including start.sh tweak worked for me while trying to install graphicsmagick. And all ways are quite hacky.

**Comment by mlecoq** [comment]
+1 I hope that it will be available soon ;)

**Comment by MasterJames** [comment]
I think it might be possible in the new version being ported to kadirahq ?  They've added hooks and custom docker images both of which I think can help with that problem. This version is not being updated or accepting PRs any more.

**Comment by timbrandin** [comment]
@MasterJames do you have a link to the work being done porting? I'd love to get that custom image's!

**Comment by bompi88** [comment]
@timbrandin The new meteor-up is [here](https://github.com/kadirahq/meteor-up).

**Comment by timbrandin** [comment]
Thx // T

**Comment by etheleon** [comment]
@bompi88 after installing graphicsmagick 

run `docker restart <containername>`

**Comment by hiteshagja** [comment]
Nothing worked above mentioned in my case so I end up using **numtel:cfs-image-resize** package. This package has no external dependency and it will work same as cfs:graphicsmagick. 

Hope this will save your days.

**Comment by timbrandin** [comment]
Yeah, same here, I use Jimp and tried out Lwip. Though I was most successful with Jimp :) https://github.com/oliver-moran/jimp
https://github.com/EyalAr/lwip

**Comment by xvendo** [comment]
This is all a huge pain. If you want to add a path env variable, or install graphicsmagick mupx is not flexible enough for an acceptable solution. There is mup that gives you the chance to add docker images. But I think you can only choose one. If you want to stay flexible, and you need to install server libraries in order to run your app, than It's really the best to do it plain https://www.digitalocean.com/community/tutorials/how-to-deploy-a-meteor-js-application-on-ubuntu-14-04-with-nginx.

It's really said that MUPX is not that flexible, because It would save you a lot of time :-(

**Comment by xvendo** [comment]
Please use MUP (the development version) there you can add images. I had the same issue and was using MUPX. It was not possible to fix the graphicsmagick integration problem. I installed latest MUP and added an docker image. It worked like a charm. You can lookup many images over here https://hub.docker.com/. Hope that helped someone. Feel free to get in touch if you need more info.

**Comment by xvendo** [comment]
I hope the man running hundreds of businesses at the same time @arunoda will find a moment to make mupx docker image aware :+1:
